frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -16,9 +16,147 @@ local C_ = _.pgettext
+    text = text:gsub("%s%s+", " ")
+    return text
+end
 
 function ReaderHighlight:init()
+    self._highlight_buttons = {
+        ["01_highlight"] = function(_self)
+            return {
+                text = _("Highlight"),
+                callback = function()
+                    _self:saveHighlight()
+                    _self:onClose()
+                end,
+                enabled = _self.hold_pos ~= nil,
+            }
+        end,
+        ["02_add_note"] = function(_self)
+            return {
+                text = _("Add Note"),
+                callback = function()
+                    _self:addNote()
+                    _self:onClose()
+                end,
+                enabled = _self.hold_pos ~= nil,
+            }
+        end,
+        ["03_copy"] = function(_self)
+            return {
+                text = C_("Text", "Copy"),
+                enabled = Device:hasClipboard(),
+                callback = function()
+                    Device.input.setClipboardText(cleanupSelectedText(_self.selected_text.text))
+                end,
+            }
+        end,
+        ["05_wikipedia"] = function(_self)
+            return {
+                text = _("Wikipedia"),
+                callback = function()
+                    UIManager:scheduleIn(0.1, function()
+                        _self:lookupWikipedia()
+                        -- We don't call _self:onClose(), we need the highlight
+                        -- to still be there, as we may Highlight it from the
+                        -- dict lookup widget
+                    end)
+                end,
+            }
+        end,
+        ["06_dictionary"] = function(_self)
+            return {
+                text = _("Dictionary"),
+                callback = function()
+                    _self:onHighlightDictLookup()
+                    -- We don't call _self:onClose(), same reason as above
+                end,
+            }
+        end,
+        ["07_translate"] = function(_self)
+            return {
+                text = _("Translate"),
+                callback = function()
+                    _self:translate(_self.selected_text)
+                    -- We don't call _self:onClose(), so one can still see
+                    -- the highlighted text when moving the translated
+                    -- text window, and also if NetworkMgr:promptWifiOn()
+                    -- is needed, so the user can just tap again on this
+                    -- button and does not need to select the text again.
+                end,
+            }
+        end,
+        ["08_search"] = function(_self)
+            return {
+                text = _("Search"),
+                callback = function()
+                    _self:onHighlightSearch()
+                    _self:onClose()
+                end,
+            }
+        end,
+    }
+
+    if not self.ui.document.info.has_pages then
+        self:addToHighlightDialog("04_view_html", function(_self)
```

Comment:
This will change the look of the highlight dialog whether in PDF or EPUB (starting from the Wikipedia button, all will be shifted, on the left side of that dialog in one format, on the right side in the other format), which is not great.
Not fond of putting it at the end, as it would shift everything next, and we might be used to finding the buttons where they currently are :/ 
We should invent something equivalent for PDF :)

Or have a bit rethinking about the button orders and positions to justify a global "pérenne" new ordering.